driver-core/src/main/java/com/datastax/driver/core/utils/UUIDs.java

```diff
@@ -108,6 +138,43 @@ private static long makeNode() {
         }
     }
 
+    private static String getProcessPiece() {
+        Integer pid = null;
+        String pidProperty = System.getProperty(PID_SYSTEM_PROPERTY);
+        if (pidProperty != null) {
+            try {
+                pid = Integer.parseInt(pidProperty);
+                LOGGER.info("PID obtained from System property {}: {}", PID_SYSTEM_PROPERTY, pid);
+            } catch (NumberFormatException e) {
+                LOGGER.warn("Incorrect integer specified for PID in System property {}: {}", PID_SYSTEM_PROPERTY, pidProperty);
+            }
+        }
+        if (pid == null && Native.isGetpidAvailable()) {
+            try {
+                pid = Native.processId();
+                LOGGER.info("PID obtained through native call to getpid(): {}", pid);
+            } catch (Exception e) {
+                LOGGER.warn("Native call to getpid() failed", e);
+            }
+        }
+        if (pid == null) {
+            try {
+                String pidJmx = ManagementFactory.getRuntimeMXBean().getName().split("@")[0];
```

Comment:
Really hard to test this as is as I need a non-POSIX system :D, I commented out the code above and it seems to work :+1: